plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts

```diff
@@ -55,8 +60,17 @@ export class GkeClusterLocator implements KubernetesClustersSupplier {
     );
   }
 
+  async getClusters(): Promise<ClusterDetails[]> {
+    if (this.clusterDetails) {
+      return this.clusterDetails;
+    }
+
+    this.clusterDetails = await this.retrieveClusters();
+    return this.clusterDetails;
+  }
+
   // TODO pass caData into the object
-  async getClusters(): Promise<GKEClusterDetails[]> {
+  async retrieveClusters(): Promise<GKEClusterDetails[]> {
```

Comment:
Does this method now need to populate `this.clusterDetails`? Otherwise will calling `retrieveClusters` actually update anything when calling `getClusters`?

Reply:
It will be called once from `getClusters` to populate the clusterDetails and will not get called again after that.  So it should be the other way around, `retrieveClusters` gets called by `getClusters` once when `getClusters` is first called.

Reply:
Gotcha, so we will need to add some logic to enable this to refresh (doesn't necessarily have to be this PR, but maybe let's add a TODO?)

Reply:
I believe the idea here is to use the schedule from backend-tasks to run the refresh(@mclarke47 example in changeset).

There's however one problem with that approach that I didn't realize initially. The scheduling functionality from backend-tasks is once every X but it's coordinated. If you run multiple instances of the kubernetes-backend only one of them will actually update at that interval. The refresh interval therefore need to be updated to something less sophisticated.

Reply:
Prior art for running things on a schedule would be https://github.com/backstage/backstage/blob/master/plugins/search-backend-node/src/runPeriodically.ts
There's no package providing this so if this would be best for this particular use case I'd suggest making a copy.

Reply:
`runPeriodically` would then become part of the API of `kubernetes-backend` though, so that KubernetesClustersSupplier implementations can use it?

Does the `ConfigClusterLocator` also need refresh support or is that handled via the existing refresh stuff that happens when `app-config.yaml` changes?

Reply:
Yeah I agree that it's unfortunate having it in the API. Do we have to expose a schedule function? I'm thinking we could start by just exposing that refresh method and let the integrator worry about how and when to call it? 

Leaving it to the user is not that great. We should probably expose `runPeriodically` from backend-tasks package anyway. Perhaps with some rename that makes it very obvious that this is executed locally and not in some distributed fashion.

I'd be nice with some suggestions from @backstage/maintainers.

Reply:
I will push something in a bit that uses `runPeriodically` in the background.  There's also a `setClusterRefreshInterval` method on the builder so we don't have to expose it.  I think that might work! :)